src/netaddress.h

```diff
@@ -107,6 +109,22 @@ class CNetAddr
         }
 
         friend class CSubNet;
+        friend class CNetAddrHash;
+};
+
+class CNetAddrHash
+{
+public:
+    size_t operator()(const CNetAddr& a) const noexcept
+    {
+        CSipHasher hasher(salt_k0, salt_k1);
+        hasher.Write(a.ip, sizeof(a.ip));
+        return (size_t)hasher.Finalize();
```

Comment:
Is this safe on 32-bit system (e.g., ARM7)?

Reply:
Yes, I think it is.

`Finalize()` returns `uint64_t` and even if `size_t` is 32 bits, then it will just take "half" of the hash. [`SaltedOutpointHasher`](https://github.com/bitcoin/bitcoin/blob/1c86ed4/src/coins.h#L108) already does such a conversion from `uint64_t` to `size_t` and it contains some comment that `size_t` must be returned.

Reply:
Any reason to not include `m_net` as well?

Reply:
No. That code is from before `m_net` existed! I added it now, thanks!